src/conversion.rs

```diff
@@ -190,15 +190,15 @@ pub trait IntoPy<T>: Sized {
 ///
 /// # fn main() -> PyResult<()> {
 /// Python::with_gil(|py| {
```

Comment:
I don't think that's coming across in the example at all, either before or after.

Looking closer, I think the text after the example is a bit outdated.

```rust
/// Note: depending on the implementation, the lifetime of the extracted result may
/// depend on the lifetime of the `obj` or the `prepared` variable.
```

There is no `prepared` variable, so it's rather unclear what is meant.

Also looking again, I'm thinking perhaps the example should be using `extract_bound` instead of `extract` as well.

I think changing to use `extract_bound` is probably in scope for this PR, but the rest should perhaps be a new issue. I'm certainly not sure how to reword the documentation here.

Reply:
Ah I see. I've proposed an alternative update to this docstring in https://github.com/PyO3/pyo3/pull/3869, what do you think of the form there?